plugins/GSdx/res/glsl/tfx_fs.glsl

```diff
@@ -842,6 +855,13 @@ void ps_main()
     return;
 #endif
 
+#if !SW_BLEND && PS_DITHER
+    ps_dither(C);
+    // Dither matrix range is [-4,3] but positive values can cause issues when
+    // software blending is not used or is unavailable.
+    C.rgb -= 3.0;
```

Comment:
So, what if C.rgb is (0,0,0)? Would there be issues in that case if you subtract 3?

Reply:
Gets clamped to zero, eventually. It's a hackfix for the squares showing up as you can see in the posted screenshots in the comments.